docs/architecture/README.md

```diff
@@ -0,0 +1,83 @@
+# NGINX Gateway Fabric Architecture
+
+This guide explains how NGINX Gateway Fabric works in simple terms.
```

Comment:
I think it could also be worth linking to https://docs.nginx.com/nginx-gateway-fabric/overview/gateway-architecture/ for more details on the architecture.